core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/PersonAccountEditViewModel.kt

```diff
@@ -0,0 +1,19 @@
+package com.ustadmobile.core.viewmodel
+
+data class PersonUsernameAndPasswordModel(
+
+    val username: String = "",
+
+    val currentPassword: String = "",
+
+    val newPassword: String = "",
+
+    val passwordConfirmed: String = "",
+)
+
+data class PersonAccountEditUiState(
+
```

Comment:
The UiState needs to have a nullable error message for each field e.g. usernameError, currentPasswordError, newPasswordError, passwordConfirmedError

It also needs to have a nullable field called "errorMessage" (e.g. in case the app cannot connect, or there is some error that does not relate directly only to one field).

It needs to have currentPasswordVisible and usernameVisible Boolean as per the view interface.

Please add only the fields mentioned above. Don't add other fields from the view interface.